### ISSUE TITLE
[Contact] Missing heading structure

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#717]
## Issue summary

The Contact Us template is missing a heading structure for its success state. Headings are crucial to conveying meaning for each section and to allow for heading navigation using assistive technology.

For example, screen reader users navigate via headings alone to gain an understanding of the current page content. Lack of heading structure will result in a frustrating user experience.

<details>
  <summary>Screenshots</summary>
  
  <img width="619" alt="Dawn contact form. Success message text is highlighted." src="https://user-images.githubusercontent.com/1392632/135497274-76add762-0a50-44a4-9b67-a8ae37d71a26.png">
>

</details

#### HTML

```html
<div
  class="form-status form-status-list form__message"
  tabindex="-1"
  autofocus
>
  <!-- Icon… -->
  Thanks for contacting us. We'll get back to you as soon as possible.
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a Contact Us template page
3. Submit the form with valid data
4. With an active screen reader, navigate via headings alone

### Expected

- Success state to include a heading.

### Actual

- Success state missing a heading.

## Recommendation

Swap the `div.form-status` element with an `h2` heading.

#### HTML

```html
<h2
  class="form-status form-status-list form__message"
  tabindex="-1"
  autofocus
>
  <!-- Icon… -->
  Thanks for contacting us. We'll get back to you as soon as possible.
</h2>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** Contact
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.10: Section Headings](https://www.w3.org/WAI/WCAG21/Understanding/section-headings.html)
- **Severity:** Low
- **Effort:** Low
